container/container.go

```diff
@@ -324,6 +324,11 @@ func (container *Container) CheckpointDir() string {
 // StartLogger starts a new logger driver for the container.
 func (container *Container) StartLogger() (logger.Logger, error) {
 	cfg := container.HostConfig.LogConfig
+
+	if cfg.Type == "none" {
+		return nil, logger.ErrReadLogsNotSupported
```

Comment:
without this, it returns a `"failed to get logging factory: %v"` because the "none" driver is not registered. I was looking if there's another location where this should/could be handled but wasn't sure.